### PR TITLE
Separate amqp_trx_plugin from amqp_trace_plugin

### PR DESCRIPTION
**Pull request by heifner**
## Change Description

- `amqp_trx_plugin` no longer depends on `amqp_trace_plugin` for sending transaction traces to AMQP. `amqp_trace_plugin` still exists and provides ability to send all transaction traces to AMQP. `amqp_trx_plugin` is setup to only send transaction traces for transactions that it processes.
- The AMQP reply-to-queue pattern is now used by `amqp_trx_plugin`. If a `reply-to` is sent with the transaction then the trace will be send to the AMQP default exchange connection with that `routing-key`. The `correlation-id` is set to the transaction id. If a `reply-to` is not specified then no trace is sent to AMQP.

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [ ] Stability bug fix

- [x] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [ ] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [x] Documentation Additions

```
  --amqp-trx-trace-reliable-mode arg (=queue)
                                        Reliable mode 'exit', exit application
                                        on any AMQP publish error.
                                        Reliable mode 'queue', queue
                                        transaction traces to send to AMQP on
                                        connection establishment.
                                        Reliable mode 'log', log an error and
                                        drop message when unable to directly
                                        publish to AMQP.

```

### CODE REVIEW

File: libraries/amqp/include/eosio/amqp/amqp_handler.hpp [hunk 1]
```diff
@@ -230,7 +230,7 @@ class amqp_handler {
          first_connect_.set();
       } );
       consumer.onReceived( [&](const AMQP::Message& message, const delivery_tag_t& delivery_tag, bool redelivered) {
-         on_consume_( delivery_tag, message.body(), message.bodySize() );
+         on_consume_( message, delivery_tag, redelivered );
```
Comment (swatanabe-b1) [inline libraries/amqp/include/eosio/amqp/amqp_handler.hpp:233]: `on_consume_t` is almost the same type as `AMQP::MessageCallback` now.  Is there a reason for forwarding the arguments through an extra lambda?

File: plugins/amqp_trx_plugin/amqp_trx_plugin.cpp [hunk 2]
```diff
@@ -187,15 +192,24 @@ void amqp_trx_plugin::set_program_options(options_description& cli, options_desc
    op("amqp-trx-ack-mode", bpo::value<ack_mode>()->default_value(ack_mode::in_block),
       "AMQP ack when 'received' from AMQP, when 'executed', or when 'in_block' is produced that contains trx.\n"
       "Options: received, executed, in_block");
+   op("amqp-trx-trace-address", bpo::value<std::string>(),
+      "AMQP address: Format: amqp://USER:PASSWORD@ADDRESS:PORT\n"
+      "Will publish to amqp-trx-trace-queue-name ('trace') queue. If not provided traces are not sent.");
+   op("amqp-trx-trace-queue-name", bpo::value<std::string>()->default_value("trace"),
+      "AMQP queue to publish transaction traces of amqp transactions.");
+   op("amqp-trx-trace-exchange", bpo::value<std::string>()->default_value(""),
+      "Existing AMQP exchange to send transaction trace messages.");
```
Comment (swatanabe-b1) [inline plugins/amqp_trx_plugin/amqp_trx_plugin.cpp:201]: I don't think that these should be options.  The trx plugin should _only_ send a trace via `reply-to`, it should _always_ send a trace when a `reply-to` is given, and it should always send the trace to the same AMQP server from which it receives transactions.  Broadcasting traces is the responsibility of the trace plugin.

**Review by swatanabe-b1 — CHANGES_REQUESTED** [review]

File: plugins/amqp_trx_plugin/amqp_trx_plugin.cpp (hunk 2 above)
Comment (heifner) [reply plugins/amqp_trx_plugin/amqp_trx_plugin.cpp:201]: This is a really good point. @arhag do you agree with this?

File: libraries/amqp/include/eosio/amqp/amqp_handler.hpp (hunk 1 above)
Comment (heifner) [reply libraries/amqp/include/eosio/amqp/amqp_handler.hpp:233]: `onReceived` takes a std::function. So no need for the extra lambda.

**Review by swatanabe-b1 — APPROVED** [review]

**Pull request merged**